### PR TITLE
storm limits ( and other options ), more telepath awareness, and cryotank / cortex integration

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/cryotank.py [hunk 1]
```diff
@@ -274,9 +289,9 @@ def last(self, name):
         tank = self.cell.init(name)
         return tank.last()
 
-    def puts(self, name, items):
+    def puts(self, name, items, seqn=None):
         tank = self.cell.init(name)
-        tank.puts(items)
+        tank.puts(items, seqn=seqn)
```
Comment (vEpiphyte) [inline synapse/cryotank.py]: we should return retn here instead of true

File: synapse/cryotank.py [hunk 2]
```diff
@@ -139,6 +148,12 @@ def puts(self, items):
                 info = {'time': tick, 'count': len(items), 'size': bytesize, 'took': took}
                 curs.put(lkey, s_msgpack.en(info), append=True)
 
+            if seqn is not None:
+                iden, offset = seqn
+                nextoff = offset + len(items)
+                self.offs.xset(xact, iden, nextoff)
+                retn = nextoff
```
Comment (vEpiphyte) [inline synapse/cryotank.py]: we should update retn regardlessof whether or not seqn is provided

File: synapse/daemon.py [hunk 3]
```diff
@@ -241,7 +241,16 @@ def share(self, name, item):
...
+
+        try:
+
+            if isinstance(item, s_telepath.Aware):
+                item.onTeleShare(self, name)
+
+            self.shared[name] = item
+
+        except Exception as e:
+            logger.exception('onTeleShare() error for: {name}')
```
Comment (vEpiphyte) [inline synapse/daemon.py]: missing `f'` formatter at the start of the string

File: synapse/lib/snap.py [hunk 4]
```diff
@@ -71,6 +74,10 @@ def fini():
 
             if self.exitok:
 
+                if self.write:
+                    if self.splices:
```
Comment (vEpiphyte) [inline synapse/lib/snap.py]: `if self.write and self.splices:`

File: synapse/tools/cryo/cat.py [hunk 5]
```diff
@@ -5,38 +5,21 @@
...
 
     pars = argparse.ArgumentParser(prog='cryo.cat', description='display data items from a cryo cell')
-    pars.add_argument('cryocell', help='The cell descriptor and cryo tank path (cell://<host:port>/<name>).')
-    pars.add_argument('--list', default=False, action='store_true', help='List tanks in the remote cell and return')
+    pars.add_argument('cryotank', help='The telepath URL for the remote cryotank.')
     pars.add_argument('--offset', default=0, type=int, help='Begin at offset index')
     pars.add_argument('--size', default=10, type=int, help='How many items to display')
-    pars.add_argument('--timeout', default=10, type=int, help='The network timeout setting')
-    pars.add_argument('--authfile', help='Path to your auth file for the remote cell')
+    # TODO: synapse.tools.cryo.list <cryocell>
```
Comment (vEpiphyte) [inline synapse/tools/cryo/cat.py]: we're gonna need this like today

File: synapse/cryotank.py [hunk 6]
```diff
@@ -69,25 +87,15 @@ def fini():
...
-            if lval is None:
-                return None
-
-            return int.from_bytes(lval, byteorder='big')
-
-    def setOffset(self, iden, offset):
+    def setOffset(self, iden, offs):
+        return self.offs.set(iden, offs)
 
+    def _xactSetOffset(self, xact, iden, offset):
```
Comment (whippit) [inline synapse/cryotank.py]: what is this for?

File: synapse/cortex.py [hunk 7]
```diff
@@ -14,6 +13,7 @@
 import synapse.lib.layer as s_layer
 import synapse.lib.syntax as s_syntax
 import synapse.lib.modules as s_modules
+import synapse.lib.threads as s_threads
```
[the remark this answers is not in the document]
Comment (invisig0th) [reply synapse/cortex.py]: hehe, i'm about to be :D lets let this one slide

File: synapse/cryotank.py (hunk 6 above)
Comment (invisig0th) [reply synapse/cryotank.py]: vestigial.  good catch!

**Pull request merged**